snapshot/snapshotter.go

```diff
@@ -282,6 +285,38 @@ func checkSnapshotterBasic(t *testing.T, snapshotter Snapshotter, work string) {
 		t.Fatal(err)
 	}
 
+	fooMode := os.FileMode(0777)
+	aMode, bMode, cMode := os.ModeDir|0755, os.ModeDir|0755, os.ModeDir|0755
+	info1 := &testutil.FileInfo{
+		BaseName: "",
+		// If you set Mode to os.ModeDir | 0777 for "", overlay fails (btrfs and naive passes)
```

Comment:
Confirmed that doing a `os.Chmod` on the directories post mount gets it to pass. My guess is the root permission is not propagated from unmount -> commit -> prepare -> mount of the new layer because the upper will get recreated with permission 0700, see https://github.com/docker/containerd/blob/master/snapshot/overlay/overlay.go#L192.